library/src/main/java/com/leinardi/android/speeddial/FabWithLabelView.java

```diff
@@ -179,21 +183,32 @@ public void setOnActionSelectedListener(OnActionSelectedListener listener) {
             getFab().setOnClickListener(new OnClickListener() {
                 @Override
                 public void onClick(View view) {
-                    mOnActionSelectedListener.onActionSelected(getSpeedDialActionItem());
+                    SpeedDialActionItem speedDialActionItem = getSpeedDialActionItem();
+                    if (mOnActionSelectedListener != null
+                            && speedDialActionItem != null) {
+                        mOnActionSelectedListener.onActionSelected(speedDialActionItem);
+                    }
                 }
             });
             getLabelBackground().setOnClickListener(new OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    if (getSpeedDialActionItem().isLabelClickable() && isLabelEnable()) {
-                        mOnActionSelectedListener.onActionSelected(getSpeedDialActionItem());
+                    SpeedDialActionItem speedDialActionItem = getSpeedDialActionItem();
+                    if (mOnActionSelectedListener != null
+                            && speedDialActionItem != null
+                            && speedDialActionItem.isLabelClickable()
+                            && isLabelEnable()) {
+                        mOnActionSelectedListener.onActionSelected(speedDialActionItem);
                     }
                 }
             });
-        } else {
+        } else
+
+        {
```

Comment:
**Reporter**: Error Prone
**Rule**: LeftCurly
**Severity**: INFO
**File**: library/src/main/java/com/leinardi/android/speeddial/FabWithLabelView.java L207





*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<885645710>*